velox/exec/tests/utils/ExpressionBuilder.h

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/parse/Expressions.h"
+
+namespace facebook::velox::expr_builder {
+
+/// Fluent Expression Builder.
+///
+/// This file contains fluent methods that make it convenient to create
+/// (untyped) expression trees. This provides similar functionality to a SQL
+/// parser, without briging dependency on external libraries or bringing leaked
+/// semantics from other systems.
+///
+/// The untyped expressions can then be turned into typed expressions ready for
+/// execution using type binding from `core::Expressions::inferTypes()`.
+///
+/// The API provided is as close to the actual expression trees as possible.
+/// Filters, arithmetics, function calls, literals, aliases, and more are
+/// supported with this API.
+///
+/// For example, to create a column reference, you can:
+///
+/// > using namespace expr_builder;
+/// > ExprPtr e = col("c0");
+///
+/// You can also use the "_c" C++ literal provided:
+///
+/// > ExprPtr e = "c0"_c;
+///
+/// Nested column references can be specified as either:
+///
+/// > ExprPtr e = col("parent", "child");
+/// > ExprPtr e = col("parent").subfield("child");
+///
+/// To debug the expression generated, you can simply:
+///
+/// > LOG(INFO) << *e;
+///
+/// Comparisons and other expressions can be fluently created using C++
```

Comment:
Would you clarify how these are translated to function calls? I assume we hard-code a mapping from C++ operator to a function name and that mapping may not work for some systems. Specifically, Spark seems to use different names for == and similar. See gluten/cpp/velox/operators/functions/SparkExprToSubfieldFilterParser.cc

Reply:
Will clarify. For now I'm just using the Presto function names "eq/neq/gt/lt/..."